sphinx/writers/latex.py

```diff
@@ -1360,25 +1376,10 @@ def depart_hlistcol(self, node):
         pass
 
     def latex_image_length(self, width_str):
-        """Convert `width_str` with rst length to LaTeX length.
-
-        This function is copied from docutils' latex writer
-        """
-        match = re.match('(\d*\.?\d*)\s*(\S*)', width_str)
-        if not match:
-            # fallback
-            return width_str
-        res = width_str
-        amount, unit = match.groups()[:2]
-        if not unit:
-            return None
-        elif unit == 'pt':
-            res = '%sbp' % amount  # convert to 'bp'
-        elif unit == "px":
-            res = "%.3f\\sphinxpxdimen" % (float(amount))
-        elif unit == "%":
-            res = "%.3f\\linewidth" % (float(amount) / 100.0)
-        return res
+        try:
+            return rstdim_to_latexdim(width_str)
+        except ValueError:
+            self.builder.warn('dimension unit %s is invalid. Ignored.')
```

Comment:
Fixed.

The pre-existing check is executed by image directive on docutils.
https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/parsers/rst/directives/__init__.py#l242
But this is a special behavior of the directive. so an image node made by 3rd party extension might  have invalid dimension units if the extension does not check.